### DISCUSSION TITLE
part-ii/2020-105/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#311] [Question comments]
Part II 2020 Number Theory: Paper 4, Section I, H 

https://questions.tripos.org/part-ii/2020-105/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

The first bit is in lectures, but the key idea is to examine $f(x)=x^{p-1}-1-\prod_{a=1}^{p-1}(x-a)$ mod $p$. This has at least $p-1$ roots mod $p$ by Fermat's little theorem, but $\deg f<p-1$, so by Lagrange $f$ is identically 0, so in fact $x^{p-1}-1\equiv\prod_{a=1}^{p-1}(x-a)$ mod $p$. Setting $x=0$ gives the result.

For the second part, note $(\mathbb Z/p\mathbb Z)^*=\langle g\rangle$ for some $g$ a primitive root mod $p$. Then $\langle g^{(p-1)k}\rangle$ is a subgroup of order $k$ with each element satisfying the congruence. So there are at most $k$ solutions mod $p$ by Lagrange, and we just found $k$ distinct solutions. So there are _precisely_ $k$ solutions.

</details>

**Comment by FTcode** [comment]
For the last part you can write $p-1 = km$ and factor $X^{p-1}-1 = (X^k - 1)(X^{(m-1)k} + X^{(m-2)k} + \ldots + X^k + 1)$. Then argue that $X^k -1$ thus splits into $k$ distinct linear factors (as $X^{p-1} -1$ does, which we showed in the previous part).
